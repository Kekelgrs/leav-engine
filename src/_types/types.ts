export interface ILabel {
    fr: string;
    en: string;
    [x: string]: string;
}

export interface ILibrary {
    id: string;
    label: ILabel;
    gqlNames: {
        query: string;
        filter: string;
        searchableFields: string;
    };
}

export type IPreview = {
    small: string;
    medium: string;
    big: string;
    pages: string;
} | null;

export interface IItem {
    [x: string]: any;
    id: string;
    label?: string;
    preview?: IPreview;
    color?: string;
    library?: {
        id: string;
        label: ILabel;
    };
}

export interface RecordIdentity_whoAmI extends IItem {}

export enum PreviewAttributes {
    small = 'small',
    medium = 'medium',
    big = 'big',
    pages = 'pages'
}

export enum PreviewSize {
    small = 'small',
    medium = 'medium',
    big = 'big'
}

export enum AvailableLanguage {
    en = 'en',
    fr = 'fr'
}

export enum FilterTypes {
    filter = 'filter',
    separator = 'separator'
}

export interface IFilterSeparatorCommon {
    key: number;
    id: string;
}

export interface IFilter extends IFilterSeparatorCommon {
    type: FilterTypes.filter;
    operator?: boolean;
    condition: ConditionFilter;
    value: string;
    attributeId: string;
    active: boolean;
    format?: AttributeFormat;
    originAttributeData?: IOriginAttributeData;
    extendedData?: IExtendedData;
    treeData?: ITreeData;
    valueSize?: number | 'auto';
}

export interface IFilterSeparator extends IFilterSeparatorCommon {
    type: FilterTypes.separator;
    active: boolean;
}

export enum AttributeFormat {
    text = 'text',
    numeric = 'numeric',
    date = 'date',
    encrypted = 'encrypted',
    boolean = 'boolean',
    extended = 'extended'
}

export enum AttributeType {
    simple = 'simple',
    simple_link = 'simple_link',
    advanced = 'advanced',
    advanced_link = 'advanced_link',
    tree = 'tree'
}

export enum OperatorFilter {
    and = 'AND',
    or = 'OR',
    openParent = 'OPEN_BRACKET',
    closeParent = 'CLOSE_BRACKET'
}

export enum ConditionFilter {
    contains = 'CONTAINS',
    notContains = 'NOT_CONTAINS',
    equal = 'EQUAL',
    notEqual = 'NOT_EQUAL',
    beginWith = 'BEGIN_WITH',
    endWith = 'END_WITH',
    greaterThan = 'GREATER_THAN',
    lessThan = 'LESS_THAN'
}

export interface IQueryFilter {
    field?: string;
    value?: any;
    condition?: ConditionFilter;
    operator?: OperatorFilter;
}

export enum OrderSearch {
    desc = 'desc',
    asc = 'asc'
}

export enum DisplayListItemTypes {
    listSmall = 'listSmall',
    listMedium = 'listMedium',
    listBig = 'listBig',
    tile = 'tile'
}

export interface IAttribute {
    id: string;
    library: string;
    type: AttributeType;
    format?: AttributeFormat;
    label: ILabel | string;
    isLink: boolean;
    isMultiple: boolean;
    linkedLibrary?: string;
    linkedTree?: string;
    originAttributeData?: IOriginAttributeData;
}

export type ExtendFormat = string | {[key: string]: ExtendFormat[]};

export interface IExtendedData {
    path: string;
    format: AttributeFormat;
}

export interface IItemsColumn {
    id: string;
    library: string;
    type: AttributeType;
    originAttributeData?: IOriginAttributeData;
    extendedData?: IExtendedData;
    treeData?: ITreeData;
}

export interface ITableHeaderOld {
    name: string;
    display: string;
    type: AttributeType;
}

export interface ITableHeader {
    title: string | any;
    type: AttributeType;
    library: string;
    dataIndex: string;
    key: string;
    width?: number;
    onHeaderCell?: (
        column: any
    ) => {
        width: number;
        onResize: any;
    };
    render: (text: string) => JSX.Element;
}

export interface IRecordEdition {
    show: boolean;
    item?: IItem;
}

export interface IAccordionActive {
    id: string;
    library: string;
    depth: number;
}

export interface IAttributesChecked {
    id: string;
    library: string;
    label: string | ILabel;
    type: AttributeType;
    depth: number;
    checked: boolean;
    originAttributeData?: IOriginAttributeData;
    extendedData?: IExtendedData;
    treeData?: ITreeData;
}

export interface IOriginAttributeData {
    id: string;
    type: AttributeType;
}

export interface ITreeData {
    attributeTreeId: string;
    libraryTypeName: string;
}

export interface IEmbeddedFields {
    id: string;
    format: AttributeFormat;
    label: ILabel;
    embedded_fields: IEmbeddedFields[];
}

export interface IGroupEmbeddedFields {
    [attributeId: string]: {
        embedded_fields: {[key: string]: IEmbeddedFields};
    };
}

export interface IAttributeSelected {
    id: string;
    library: string;
    originAttributeData?: IOriginAttributeData;
    extendedData?: IExtendedData;
    treeData?: ITreeData;
}

export interface ITree {
    id: string;
    label: ILabel;
    libraries: {
        id: string;
        label: ILabel;
    }[];
}

export interface INavigationPath {
    id: string;
    library: string;
    label?: string | null;
}

export enum NotificationType {
    basic = 'basic',
    success = 'success',
    warning = 'warning',
    error = 'error'
}

export enum NotificationPriority {
    low = 'low',
    medium = 'medium',
    high = 'high'
}

export interface IBaseNotification {
    content: string;
    type: NotificationType;
}

export interface INotification extends IBaseNotification {
    time?: number;
    priority?: NotificationPriority;
}

export type ILang = string[];
