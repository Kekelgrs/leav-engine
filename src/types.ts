import {Channel} from 'amqplib';

export interface IFullTreeContent {
    [index: number]: {order: number; record: any; children: IFullTreeContent[]};
}

export interface IAmqpParams {
    channel?: Channel;
    exchange?: string;
    routingKey?: string;
}

export interface IParams {
    rootPath: string;
    rootKey: string;
    verbose: boolean;
    amqp?: IAmqpParams;
}

export interface IMessageSend {
    event: string;
    time: number;
    pathAfter: string;
    pathBefore: string;
    inode: number;
    isDirectory: boolean;
    rootKey: any;
}
