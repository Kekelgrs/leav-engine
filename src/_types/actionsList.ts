import {description} from 'joi';
import {IAttribute} from './attribute';
import libraryDomain from 'domain/libraryDomain';

export enum ActionsListEvents {
    SAVE_VALUE = 'saveValue',
    DELETE_VALUE = 'deleteValue',
    GET_VALUE = 'getValue'
}

export enum ActionsListIOTypes {
    STRING = 'string',
    NUMBER = 'number',
    OBJECT = 'object',
    BOOLEAN = 'boolean'
}

export type ActionsListValueType = string | number | boolean | {};

// TODO: should be able to use [event: ActionsListEvents] as from TS v2.9
export interface IActionsListConfig {
    [ActionsListEvents.SAVE_VALUE]?: [IActionsListSavedAction];
    [ActionsListEvents.DELETE_VALUE]?: [IActionsListSavedAction];
    [ActionsListEvents.GET_VALUE]?: [IActionsListSavedAction];
}

export interface IActionsListContext {
    attribute?: IAttribute;
    library?: string;
    recordId?: number;
}

export interface IActionsListParams {
    [name: string]: any;
}

export interface IActionsListFunction {
    name: string;
    description: string;
    inputTypes: ActionsListIOTypes[];
    outputTypes: ActionsListIOTypes[];
    params?: [{name: string; type: string; description: string}];
    action: (value: ActionsListValueType, params: IActionsListParams, ctx: IActionsListContext) => ActionsListValueType;
}

export interface IActionsListSavedAction {
    name: string;
    isSystem: boolean;
    params?: [{name: string; value: string}];
}
