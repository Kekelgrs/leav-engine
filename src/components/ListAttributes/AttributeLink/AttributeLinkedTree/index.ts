import AttributeLinkedTree from './AttributeLinkedTree';
export default AttributeLinkedTree;
