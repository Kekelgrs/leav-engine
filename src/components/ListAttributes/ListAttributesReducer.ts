// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {IAccordionActive, IAttribute, IAttributesChecked, IAttributeSelected} from './../../_types/types';

export enum ListAttributeReducerActionTypes {
    SET_ATTRIBUTE_SELECTED = 'SET_ATTRIBUTE_SELECTED',
    SET_ATTRS_CHECKED = 'SET_ATTRS_CHECKED',
    SET_LANG = 'SET_LANG',
    SET_CURRENT_ACCORDION = 'SET_CURRENT_ACCORDION',
    SET_NEW_ATTRIBUTES = 'SET_NEW_ATTRIBUTES',
    RESET = 'RESET'
}

export interface ListAttributeState {
    lang: any;
    accordionsActive: IAccordionActive[];
    newAttributes: IAttribute[];
    attributesChecked: IAttributesChecked[];
    useCheckbox?: boolean;
    attributeSelected?: IAttributeSelected;
    changeSelected?: (attId: IAttributeSelected) => void;
}

export const ListAttributeInitialState: ListAttributeState = {
    attributesChecked: [],
    lang: null,
    accordionsActive: [],
    newAttributes: []
};

export type ListAttributeReducerAction =
    | {
          type: ListAttributeReducerActionTypes.SET_ATTRIBUTE_SELECTED;
          attributeSelected: IAttributeSelected;
      }
    | {
          type: ListAttributeReducerActionTypes.SET_ATTRS_CHECKED;
          attributesChecked: IAttributesChecked[];
      }
    | {
          type: ListAttributeReducerActionTypes.SET_LANG;
          lang: any;
      }
    | {
          type: ListAttributeReducerActionTypes.SET_CURRENT_ACCORDION;
          accordionsActive: IAccordionActive[];
      }
    | {
          type: ListAttributeReducerActionTypes.SET_NEW_ATTRIBUTES;
          newAttributes: IAttribute[];
      }
    | {
          type: ListAttributeReducerActionTypes.RESET;
      };

export const ListAttributeReducer = (
    state: ListAttributeState,
    action: ListAttributeReducerAction
): ListAttributeState => {
    switch (action.type) {
        case ListAttributeReducerActionTypes.SET_ATTRIBUTE_SELECTED:
            return {...state, attributeSelected: action.attributeSelected};
        case ListAttributeReducerActionTypes.SET_ATTRS_CHECKED:
            return {...state, attributesChecked: action.attributesChecked};
        case ListAttributeReducerActionTypes.SET_LANG:
            return {...state, lang: action.lang};
        case ListAttributeReducerActionTypes.SET_CURRENT_ACCORDION:
            return {...state, accordionsActive: action.accordionsActive};
        case ListAttributeReducerActionTypes.SET_NEW_ATTRIBUTES:
            return {...state, newAttributes: action.newAttributes};
        case ListAttributeReducerActionTypes.RESET:
            return ListAttributeInitialState;
        default:
            return state;
    }
};
