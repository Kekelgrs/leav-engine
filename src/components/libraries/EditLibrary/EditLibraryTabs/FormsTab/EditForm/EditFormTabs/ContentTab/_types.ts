import {DragObjectWithType} from 'react-dnd';
import {GET_ATTRIBUTES_attributes_list} from '../../../../../../../../_gqlTypes/GET_ATTRIBUTES';
import {FormElementTypes} from '../../../../../../../../_gqlTypes/globalTypes';
import {IKeyValue} from '../../../../../../../../_types/shared';

export const PLACEHOLDER_ID = '__placeholder__';

export enum UIElementTypes {
    DIVIDER = 'divider',
    FIELDS_CONTAINER = 'fields_container',
    TEXT_BLOCK = 'text_block',
    TABS = 'tabs'
}

export enum FieldTypes {
    TEXT_INPUT = 'input_field',
    DATE = 'date',
    CHECKBOX = 'checkbox',
    ENCRYPTED = 'encrypted',
    DROPDOWN = 'dropdown'
}

export enum DraggableElementTypes {
    RESERVE_LAYOUT_ELEMENT = 'RESERVE_LAYOUT_ELEMENT',
    ATTRIBUTE = 'ATTRIBUTE',
    FORM_ELEMENT = 'FORM_ELEMENT'
}

export interface IFormElement {
    id: string;
    order: number;
    type: FormElementTypes;
    uiElement: IUIElement;
    containerId: string;
    settings?: IKeyValue<any>;
    herited?: boolean;
}

export interface IUIElement {
    type: UIElementTypes | FieldTypes;
    component: JSX.Element;
    canDrop: (dropCandidate: IFormElement) => boolean;
    settings?: string[];
}

export interface IFormElementProps<T extends object> {
    elementData?: IFormElement;
    settings: T;
}

export interface ICommonFieldsSettings {
    label?: string;
    attribute?: GET_ATTRIBUTES_attributes_list;
}

export interface IFormElementPos {
    order: number;
    containerId: string;
}

export interface IFormBuilderDragObject<T extends IUIElement | IFormElement> extends DragObjectWithType {
    type: DraggableElementTypes;
    element: T;
    index: number;
    currentPos?: IFormElementPos;
    originPos?: IFormElementPos;
    dropAtPos?: IFormElementPos;
}
