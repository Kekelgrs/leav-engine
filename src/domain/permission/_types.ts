import {
    AppPermissionsActions,
    IPermissionsTreeTarget,
    LibraryPermissionsActions,
    PermissionsActions,
    PermissionTypes
} from '_types/permissions';
import {IQueryInfos} from '_types/queryInfos';
import {ITreeNode} from '_types/tree';

export interface IPermissionTarget {
    attributeId?: string;
    recordId?: string;
}

export interface IGetSimplePermissionsParams {
    type: PermissionTypes;
    applyTo: string | null;
    action: PermissionsActions;
    usersGroupId: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export interface IGetPermissionsByActionsParams {
    type: PermissionTypes;
    applyTo: string | null;
    actions: PermissionsActions[];
    usersGroupId: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export type PermByActionsRes = {[name: string]: boolean | null} | null;

export interface IGetPermissionByUserGroupsParams {
    type: PermissionTypes;
    action: PermissionsActions;
    userGroupsPaths: ITreeNode[][];
    applyTo?: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export interface IGetAppPermissionParams {
    action: AppPermissionsActions;
    userId: string;
    ctx: IQueryInfos;
}

export interface IGetHeritedAppPermissionParams {
    action: AppPermissionsActions;
    userGroupId: string;
    ctx: IQueryInfos;
}

export interface IGetLibraryPermissionParams {
    action: LibraryPermissionsActions;
    libraryId: string;
    userId: string;
    ctx: IQueryInfos;
}

export interface IGetHeritedLibraryPermissionParams {
    action: LibraryPermissionsActions;
    libraryId: string;
    userGroupId: string;
    ctx: IQueryInfos;
}

export interface IGetHeritedPermissionsParams {
    type: PermissionTypes;
    applyTo: string;
    action: PermissionsActions;
    userGroupId: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export interface IIsAllowedParams {
    type: PermissionTypes;
    action: PermissionsActions;
    userId: string;
    applyTo?: string;
    target?: IPermissionTarget;
    ctx: IQueryInfos;
}
