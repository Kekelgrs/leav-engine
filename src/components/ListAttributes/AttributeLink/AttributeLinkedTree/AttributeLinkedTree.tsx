import {useLazyQuery} from '@apollo/client';
import React, {useEffect, useState} from 'react';
import {Button, Checkbox, Icon, Loader, Radio} from 'semantic-ui-react';
import styled from 'styled-components';
import {getTreeAttributesQuery} from '../../../../queries/trees/getTreeAttributesQuery';
import {attributeUpdateSelection, checkTypeIsLink, localizedLabel} from '../../../../utils';
import {IAccordionActive, IAttribute, IOriginAttributeData} from '../../../../_types/types';
import {ListingAttributes} from '../../ListAttributes';
import {
    ListAttributeReducerAction,
    ListAttributeReducerActionTypes,
    ListAttributeState
} from '../../ListAttributesReducer';
import {
    CustomAccordion,
    CustomAccordionContent,
    CustomAccordionTitle,
    RowAttribute,
    SmallText,
    TextAttribute,
    WrapperAttribute,
    WrapperContentAttribute
} from '../../StyledComponents';

const LibraryName = styled.div`
    font-weight: 300;
    opacity: 0.3;
    width: 100%;
    justify-content: space-around;
    display: flex;
`;

interface ITreeLinkedAttribute {
    attributes: IAttribute[];
    library: string;
    libraryTypeName: string;
}

interface IAttributeLinkedTreeProps {
    attribute: IAttribute;
    currentAccordion?: IAccordionActive;
    changeCurrentAccordion: () => void;
    stateListAttribute: ListAttributeState;
    dispatchListAttribute: React.Dispatch<ListAttributeReducerAction>;
    depth: number;
    isChecked: boolean;
    originAttributeData?: IOriginAttributeData;
}

function AttributeLinkedTree({
    attribute,
    currentAccordion,
    changeCurrentAccordion,
    stateListAttribute,
    dispatchListAttribute,
    depth,
    isChecked,
    originAttributeData
}: IAttributeLinkedTreeProps): JSX.Element {
    const [linkedAttributes, setLinkedAttributes] = useState<ITreeLinkedAttribute[]>([]);

    const [getLinkedAttributes, {called, loading, data, error}] = useLazyQuery(getTreeAttributesQuery, {
        variables: {
            treeId: attribute.linkedTree
        }
    });

    useEffect(() => {
        if (currentAccordion?.id === attribute.id && !called) {
            getLinkedAttributes();
        }

        if (called && !loading) {
            const newLinkedAttributes: ITreeLinkedAttribute[] = data?.trees?.list[0]?.libraries?.map(library => {
                return {
                    library: library.id,
                    libraryTypeName: library.gqlNames.type,
                    attributes: library.attributes.map(attr => ({
                        id: attr.id,
                        type: attr.type,
                        format: attr.format,
                        label: attr.label,
                        isLink: checkTypeIsLink(attr.type),
                        isMultiple: attr.multiple_values,
                        linkedLibrary: attr.linked_library,
                        linkedTree: attr.linked_tree,
                        library: library.id,
                        originAttributeData: {id: attribute.id, type: attribute.type}
                    }))
                };
            });

            setLinkedAttributes(newLinkedAttributes);

            const newAttributes: IAttribute[] = newLinkedAttributes
                .map(lib => {
                    return lib.attributes;
                })
                .flat();

            dispatchListAttribute({
                type: ListAttributeReducerActionTypes.SET_NEW_ATTRIBUTES,
                newAttributes
            });
        }
    }, [
        loading,
        called,
        data,
        currentAccordion,
        attribute,
        getLinkedAttributes,
        setLinkedAttributes,
        dispatchListAttribute
    ]);

    if (error) {
        return <>error</>;
    }

    const isAccordionActive = currentAccordion && currentAccordion?.depth === depth;

    const handleClick = () => {
        const newAttributesChecked = attributeUpdateSelection({
            attribute,
            attributesChecked: stateListAttribute.attributesChecked,
            useCheckbox: !!stateListAttribute.useCheckbox,
            depth,
            originAttributeData
        });

        dispatchListAttribute({
            type: ListAttributeReducerActionTypes.SET_ATTRS_CHECKED,
            attributesChecked: newAttributesChecked
        });
    };

    const handleRadioChange = () => {
        if (stateListAttribute.changeSelected) {
            stateListAttribute.changeSelected(attribute.id);
        }
    };

    return (
        <WrapperAttribute>
            <CustomAccordion>
                <CustomAccordionTitle active={isAccordionActive} index={attribute.id}>
                    <Button
                        icon={isAccordionActive ? 'angle up' : 'angle down'}
                        loading={called && loading}
                        onClick={changeCurrentAccordion}
                        compact
                        size="mini"
                        circular
                    />
                    <RowAttribute onClick={handleClick}>
                        <TextAttribute>
                            {stateListAttribute.lang && localizedLabel(attribute.label, stateListAttribute.lang) ? (
                                <span>
                                    {localizedLabel(attribute.label, stateListAttribute.lang)}
                                    <SmallText>{attribute.id}</SmallText>
                                </span>
                            ) : (
                                attribute.id
                            )}
                            <span>
                                <Icon name="tree" />
                            </span>
                        </TextAttribute>
                        {stateListAttribute.useCheckbox && <Checkbox checked={isChecked} onChange={handleClick} />}
                    </RowAttribute>

                    {stateListAttribute.attributeSelection && (
                        <Radio
                            checked={stateListAttribute.attributeSelection === attribute.id}
                            onChange={handleRadioChange}
                        />
                    )}
                </CustomAccordionTitle>

                <CustomAccordionContent active={isAccordionActive}>
                    {loading ? (
                        <Loader />
                    ) : (
                        <WrapperContentAttribute>
                            {linkedAttributes.map(linkedAttribute => (
                                <div key={linkedAttribute.library}>
                                    <LibraryName>{linkedAttribute.library}</LibraryName>
                                    <ListingAttributes
                                        attributes={linkedAttribute.attributes}
                                        stateListAttribute={stateListAttribute}
                                        dispatchListAttribute={dispatchListAttribute}
                                        depth={depth + 1}
                                        originAttributeData={{id: attribute.id, type: attribute.type}}
                                        treeData={{
                                            attributeTreeId: attribute.id,
                                            libraryTypeName: linkedAttribute.libraryTypeName
                                        }}
                                    />
                                </div>
                            ))}
                        </WrapperContentAttribute>
                    )}
                </CustomAccordionContent>
            </CustomAccordion>
        </WrapperAttribute>
    );
}

export default AttributeLinkedTree;
