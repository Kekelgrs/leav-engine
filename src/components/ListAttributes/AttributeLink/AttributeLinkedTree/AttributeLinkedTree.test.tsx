import {mount} from 'enzyme';
import React from 'react';
import {act} from 'react-dom/test-utils';
import {AttributeType, IAttribute} from '../../../../_types/types';
import {ListAttributeInitialState} from '../../ListAttributesReducer';
import AttributeLinkedTree from './AttributeLinkedTree';

describe('AttributeLinkedTree', () => {
    const mockAttribute: IAttribute = {
        id: 'test',
        library: 'testLib',
        type: AttributeType.tree,
        label: 'testLabel',
        isLink: false,
        isMultiple: false,
        linkedTree: 'testLinkedTree'
    };

    test('Snapshot test', async () => {
        let comp: any;

        await act(async () => {
            comp = mount(
                <AttributeLinkedTree
                    attribute={mockAttribute}
                    changeCurrentAccordion={jest.fn()}
                    stateListAttribute={ListAttributeInitialState}
                    dispatchListAttribute={jest.fn()}
                    depth={0}
                    isChecked={true}
                />
            );
        });

        expect(comp).toMatchSnapshot();
    });
});
