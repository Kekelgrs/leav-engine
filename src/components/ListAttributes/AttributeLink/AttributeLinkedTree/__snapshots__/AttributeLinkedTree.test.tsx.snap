// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeLinkedTree Snapshot test 1`] = `ReactWrapper {}`;
